master/static/srv/shell-entrypoint.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -e
+
+export PATH="/run/determined/pythonuserbase/bin:$PATH"
+
+# Unlike trial and notebook entrypoints, the HOME directory does not need to be
+# modified in this entrypoint because the HOME in the user's ssh session is set
+# by sshd at a later time.
+
+python3.6 -m pip install --user /opt/determined/wheels/determined*.whl
+
+# Prepend each key in authorized_keys with a set of environment="KEY=VALUE"
+# options to inject the entire docker environment into the eventual ssh
+# session via an options in the authorized keys file.  See syntax described in
+# `man 8 sshd`.  Normal ssh mechanisms for overriding variables as part of the
+# protocol (like TERM or LANG) will take precedence, as will normal mechanisms
+# like a ~/.bashrc.  The purpose of this is to honor the environment variable
+# settings as they are set for experiment or notebook configs, while still
+# allowing customizations via normal ssh mechanisms.
+#
+# After openssh 8+ is the only version of openssh supported (that is, after we
+# only support ubuntu >= 20.04), we can use the more obvious SetEnv option and
+# skip this awkwardness.
+python3.6 << EOF
+import os
+import shlex
+
+# Build a string with each environment variable as an option
+opts = ",".join(f'environment="{k}={shlex.quote(v)}"' for k, v in os.environ.items())
```

Comment:
blocking: This is incorrect -- the parsing is different from shell parsing. For example, if the agent id is `"`, the result is `environment="DET_AGENT_ID='"'"`, which breaks SSH and includes extra single quotes.

Reply:
I just took the shell escaping you suggested